mpxe/harness/pm.py

```diff
@@ -30,11 +32,17 @@ def __init__(self):
         self.poll_thread.start()
         self.can = canio.CanIO()
 
-    def start(self, name, sim=None):
+    def start(self, name, sim=None, init_conds=None):
         if name not in self.proj_name_list:
             raise ValueError('invalid project "{}": expected something from projects directory')
         proj = project.Project(name, sim or Sim())
         self.fd_to_proj[proj.popen.stdout.fileno()] = proj
+        if init_conds:
+            for update in init_conds:
+                proj.write_store(update)
+                sleep(0.1)
```

Comment:
Yeah, for whatever reason both messages won't be either sent or received? When I was testing without the sleep, the C-side would only receive the command message, and not the init conditions. Not sure why this is an issue though, I would think that it should be fine without it

Reply:
I think this is because after we removed the ctop_fifo, there's no more lock on python to wait until C starts polling. Can you add something to synchronize? i.e. after C starts the poll thread, send a message to python saying C is ready, and in python wait for that message before sending the init conds. I think it makes sense to do this via a lock in this python file, and command handling in python similar to how it's done in C in the poll thread, but I'm sure there's plenty of other solutions you could explore.

Reply:
I added a locking mechanism but it didn't resolve the issue. It's in the latest commit so you can have a look. I don't think it has to do with the polling thread not being available, I think it has to do with the behaviour of the pipe itself?

 The sleep() needs to happen between sequential messages (including more than one init cond) for all messages to be received properly. Otherwise some get lost. If I remove the sleeps, for whatever reason it will still always receive the command msg, but it will not receive any initial conditions.